### PR TITLE
'amount' helper should handle zero correctly [PURCHASE-995]

### PR DESCRIPTION
**Pull request by ds300**
https://artsyproduct.atlassian.net/browse/PURCHASE-995

Need to check that we're not breaking usages of this before merging (I'm checking in reaction, emission, and volt):

 ### BuyOrder/OfferOrder
  - [x] itemsTotal
  - [x] sellerTotal
  - [x] commissionFee
  - [x] totalListPrice
  - [x] buyerTotal
  - [x] taxTotal
  - [x] shippingTotal
  - [x] transactionFee
 ### LineItem
  - [x] shippingTotal
  - [x] listPrice
  - [x] commissionFee
 ### Offer
  - [x] amount
  - [x] taxTotal
  - [x] shippingTotal
  - [x] buyerTotal
 ### AnalyticsHistogramBin
  - [x] minPrice
  - [x] maxPrice
 ### HighestBid
  - [x] amount
 ### Invoice
  - [x] total
 ### BuyersPremium
  - [x] amount
 ### SaleArtworkHighestBid
  - [x] amount

### CODE REVIEW

**Comment by sweir27** [comment]
_this is so exciting_

File: src/schema/artwork/index.ts [hunk 1]
```diff
@@ -560,11 +560,12 @@ export const ArtworkType = new GraphQLObjectType<any, ResolverContext>({
           )
             return "Free shipping worldwide"
           var domesticShipping = amount(
-            ({ domestic_shipping_fee_cents }) => domestic_shipping_fee_cents
+            ({ domestic_shipping_fee_cents }) =>
+              domestic_shipping_fee_cents || null
```
Comment (sweir27) [inline src/schema/artwork/index.ts]: Why do we need the added `|| null` here? Don't we already return `null` from the money helper if this is undefined?

**Comment by sweir27** [comment]
Just verifying-- did you take a look at deployed clients (i.e. emission) to ensure they're not depending on `null` values here?

(a la https://github.com/artsy/metaphysics/pull/1325#issuecomment-428853308)

File: src/schema/artwork/index.ts (hunk 1 above)
Comment (ds300) [reply src/schema/artwork/index.ts]: In case domestic_shipping_free_cents is 0, to preserve the old behaviour.

**Comment by ds300** [comment]
😬 no I didn't check the deployed version, only master.

**Comment by ds300** [comment]
Alright @sweir27 checked some older versions of emission and the usage is all safe. So that's emission, reaction, metaphysics, and volt checked. I don't think positron uses any of these fields. Can you think of anywhere else to check? If not I think we're good to go :+1: :tada:

**Comment by sweir27** [comment]
Thank you! I think Eigen, Force, and Prediction might be the last ones? (I don't think Eigen consumes metaphysics _much_ but it does some).

Also thank you! This is tedious but so so important 🙌 🙌

**Comment by ds300** [comment]
Ah yes forgot that I also checked prediction and it's 👍 but not force or eigen.

Just checked them both now and we're good :+1: :+1:

**Comment by sweir27** [comment]
Woooo! This looks great.

I'll assign it to you since once this is out in staging we should be sure to spot-check a few places!

**Pull request merged**